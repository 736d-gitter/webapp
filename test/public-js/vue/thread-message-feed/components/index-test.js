'use strict';

const mount = require('../../vuex-mount');
const {
  default: Index
} = require('../../../../../public/js/vue/thread-message-feed/components/index.vue');

describe('thread-message-feed index', () => {
  const addDefaultUser = state => (state.user = { displayName: 'John Smith' });
  const addParentMessage = state => {
    const parentMessage = {
      id: '5d147ea84dad9dfbc522317a'
    };
    state.threadMessageFeed.parentMessage = parentMessage;
  };

  it('closed - matches snapshot', async () => {
    const { wrapper } = mount(Index, {}, store => {
      store.state.threadMessageFeed.isVisible = false;
      // Further state changes shouldn't be necessary but for some reason the `mount` helper
      // tries to render the v-if hidden child components as well and causes Vue warnings
      addParentMessage(store.state);
      addDefaultUser(store.state);
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('opened - matches snapshot', () => {
    const { wrapper } = mount(Index, {}, store => {
      addParentMessage(store.state);
      addDefaultUser(store.state);
      store.state.threadMessageFeed.isVisible = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('dark theme - matches snapshot', () => {
    const { wrapper } = mount(Index, {}, store => {
      addParentMessage(store.state);
      addDefaultUser(store.state);
      store.state.threadMessageFeed.isVisible = true;
      store.state.darkTheme = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
