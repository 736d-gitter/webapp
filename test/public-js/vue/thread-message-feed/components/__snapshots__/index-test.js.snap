// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed index closed - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root"
>
  <!---->
</div>
`;

exports[`thread-message-feed index dark theme - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened dark-theme"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      class="content"
    >
      <div
        class="chat-messages"
      >
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
      </div>
       
      <chat-input-stub
        roommember="true"
        thread="true"
        user="[object Object]"
      />
    </section>
  </section>
</div>
`;

exports[`thread-message-feed index missing parent message - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      class="content"
    >
      <span
        class="error-text"
      >
        
        Error: The message for this thread is unavailable. It was probably deleted.
      
      </span>
    </section>
  </section>
</div>
`;

exports[`thread-message-feed index opened - matches snapshot 1`] = `
<div
  class="js-thread-message-feed-root root opened"
>
  <section
    class="body"
  >
    <thread-header-stub />
     
    <section
      class="content"
    >
      <div
        class="chat-messages"
      >
        <chat-item-stub
          message="[object Object]"
          usecompactstyles="true"
        />
      </div>
       
      <chat-input-stub
        roommember="true"
        thread="true"
        user="[object Object]"
      />
    </section>
  </section>
</div>
`;
